### PR TITLE
Chapter 10: Updating, showing and deleting users

### PR DESCRIPTION
**Pull request by rashidfirhat98**

### CODE REVIEW

**Comment by rashidfirhat98** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -2,11 +2,11 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by email: params[:session][:email].downcase
-    if user&.authenticate params[:session][:password]
-      log_in user
-      params[:session][:remember_me] == Settings.remember_me ? remember(user) : forget(user)
-      redirect_to user
+    @user = User.find_by email: params[:session][:email].downcase
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]: We dont need @ here

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: %i(show new create)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
+
+  def index
+    @users = User.page(params[:page])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    @users = User.page(params[:page]).per <number>
```
Dont forget to move number to settings file

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -19,8 +27,47 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by(id: params[:id])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -19,8 +27,47 @@ def create
...
+    if @user.update_attributes(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else
+      render "edit"
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
      render :edit
``` 
and add flash for this case

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -19,8 +27,47 @@ def create
...
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    @user = User.find_by(id: params[:id])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -19,8 +27,47 @@ def create
...
+    return if current_user?(@user)
+    flash[:danger] = "Unauthorized entry"
+    redirect_to(root_url)
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    return if current_user? @user
    
    flash[:danger] = "Unauthorized entry"
    redirect_to root_url
```

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (rashidfirhat98) [reply app/controllers/sessions_controller.rb]: done

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (rashidfirhat98) [reply app/controllers/users_controller.rb]: done sir

**Comment by rashidfirhat98** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**